devtools/cli/src/main/java/io/quarkus/cli/core/ExecuteUtil.java

```diff
@@ -0,0 +1,159 @@
+package io.quarkus.cli.core;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.InputStreamReader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.Optional;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+
+import org.apache.maven.shared.invoker.DefaultInvocationRequest;
+import org.apache.maven.shared.invoker.DefaultInvoker;
+import org.apache.maven.shared.invoker.InvocationRequest;
+import org.apache.maven.shared.invoker.InvocationResult;
+import org.apache.maven.shared.invoker.Invoker;
+
+import io.quarkus.devtools.project.BuildTool;
+import io.quarkus.devtools.project.QuarkusProject;
+import picocli.CommandLine;
+
+public class ExecuteUtil {
+
+    public static void executeGradle(File projectDirectory, String buildTarget, CommandLine cl)
+            throws Exception {
+        String gradleExecutable = findExecutable("gradle");
+        if (gradleExecutable == null) {
+            cl.getErr().println("Unable to find the gradle executable, is it in your path?");
+        } else {
+            gradleExecutable += File.separator + "bin" + File.separator + "gradle";
+
+            Process process = new ProcessBuilder()
+                    .command(gradleExecutable, buildTarget)
+                    .directory(projectDirectory)
+                    .start();
+
+            BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream()));
+
+            String line;
+            while ((line = reader.readLine()) != null) {
+                cl.getOut().println(line);
+            }
+
+            int exit = process.waitFor();
+            if (exit != 0)
+                cl.getErr().println("Build failed.");
+        }
+    }
+
+    public static void executeMaven(File projectDirectory, String buildTarget, CommandLine cl) throws Exception {
+        String mvnPath = findExecutable("mvn");
+        System.setProperty("maven.home", mvnPath);
+
+        InvocationRequest request = new DefaultInvocationRequest();
+        request.setPomFile(new File(projectDirectory.getAbsolutePath() + File.separatorChar + "pom.xml"));
+        request.setGoals(Collections.singletonList(buildTarget));
+
+        Invoker invoker = new DefaultInvoker();
+
+        InvocationResult result = invoker.execute(request);
+
+        if (result.getExitCode() != 0) {
```

Comment:
when no no `mvnw` nor `mvn` in path this seem to just fail silently. no error output, no nothing.

Reply:
stderr wasn't being outputted.  All this code has changed and this is fixed.